builders/interact/ethereum-api/precompiles/call-permit.md

```diff
@@ -0,0 +1,275 @@
+---
+title:  Call Permit Precompile Contract
+description: Learn how to use the Call Permit Precompile contract on Moonbeam to sign a permit for any EVM call that can be dispatched by anyone or any smart contract.
+keywords: solidity, ethereum, call permit, permit, gasless transaction, moonbeam, precompiled, contracts
+---
+
+# Interacting with the Call Permit Precompile
+
+## Introduction {: #introduction }
+
+The Call Permit Precompile on Tanssi EVM ContainerChains allows a user to sign a permit, an [EIP-712](https://eips.ethereum.org/EIPS/eip-712){target=\_blank} signed message, for any EVM call and it can be dispatched by anyone or any smart contract. It is similar to the Permit Signing of ERC-20 approvals introduced in [EIP-2612](https://eips.ethereum.org/EIPS/eip-2612){target=\_blank}, except it applies to any EVM call instead of only approvals.
+
+When the call permit is dispatched, it is done so on behalf of the user who signed the permit and the user or contract that dispatches the permit is responsible for paying transaction fees. As such, the precompile can be used to perform gas-less transactions.
+
+For example, Alice signs a call permit and Bob dispatches it and performs the call on behalf of Alice. Bob pays for the transaction fees and as such, Alice doesn't need to have any of the native currency to pay for the transaction, unless the call includes a transfer.
+
+The Call Permit Precompile is located at the following address:
+
+```text
+{{ networks.dancebox.precompiles.call_permit }}
+```
+
+!!! note
+    There can be some unintended consequences when using precompiles. Tanssi's Batch Precompile is derived from Moonbeam's, and as such, please familiarize yourself with [Moonbeam's Precompile Security Considerations](https://docs.moonbeam.network/builders/get-started/eth-compare/security/){target=\_blank}. 
+
+
+## The Call Permit Solidity Interface {: #the-call-permit-interface }
+
+[`CallPermit.sol`](https://github.com/moondance-labs/tanssi/blob/master/test/contracts/solidity/CallPermit.sol){target=\_blank} is a Solidity interface that allows developers to interact with the precompile's three methods.
+
+??? code "CallPermit.sol"
+
+    ```solidity
+    --8<-- 'code/builders/interact/ethereum-api/precompiles/call-permit/call-permit.sol'
+    ```
+
+The interface includes the following functions:
+
+???+ function "**dispatch**(*address* from, *address* to, *uint256* value, *bytes* data, *uint64[]* gaslimit, *uint256* deadline, *uint8* v, *bytes32* r, *bytes32* s) — dispatches a call on the behalf of another user with a EIP-712 permit. This function can be called by anyone or any smart contract. The transaction will revert if the permit is not valid or if the dispatched call reverts or errors (such as out of gas). If successful, the nonce of the signer is increased to prevent this permit to be replayed."
```

Comment:
```suggestion
???+ function "**dispatch**(*address* from, *address* to, *uint256* value, *bytes* data, *uint64[]* gaslimit, *uint256* deadline, *uint8* v, *bytes32* r, *bytes32* s) — dispatches a call on the behalf of another user with a EIP-712 permit. This function can be called by anyone or any smart contract. The transaction will revert if the permit is not valid or if the dispatched call reverts or errors (such as out of gas). If successful, the nonce of the signer is increased to prevent this permit to be replayed"
```